compose/cli/main.py

```diff
@@ -328,20 +328,20 @@ def run(self, project, options):
         if options['--user']:
             container_options['user'] = options.get('--user')
 
+        if not options['--service-ports']:
+            container_options['ports'] = []
+
         container = service.create_container(
             one_off=True,
             insecure_registry=insecure_registry,
             **container_options
         )
 
-        service_ports = None
-        if options['--service-ports']:
-            service_ports = service.options['ports']
         if options['-d']:
-            service.start_container(container, ports=service_ports, one_off=True)
+            service.start_container(container)
             print(container.name)
         else:
-            service.start_container(container, ports=service_ports, one_off=True)
+            service.start_container(container)
```

Comment:
minor (since it's existing), this is being done in both branches of the is/else, so it should really happen before the branch.